cubical/data_handler.py

```diff
@@ -1030,15 +1030,77 @@ def __init__(self, ms_name, data_column, output_column=None, output_model_column
         _spwtab = pt.table(self.ms_name + "::SPECTRAL_WINDOW", ack=False)
         _poltab = pt.table(self.ms_name + "::POLARIZATION", ack=False)
         _ddesctab = pt.table(self.ms_name + "::DATA_DESCRIPTION", ack=False)
+        _obstab = pt.table(self.ms_name + "::OBSERVATION", ack=False)
         _feedtab = pt.table(self.ms_name + "::FEED", ack=False)
+        _tab = pt.table(self.ms_name + "::FEED", ack=False)
 
         self.ctype = np.complex128 if double_precision else np.complex64
         self.ftype = np.float64 if double_precision else np.float32
         self.ncorr = _poltab.getcol("NUM_CORR")[0]
         self.nants = _anttab.nrows()
-
+        
+        # antenna fields to be used when writing gain tables
+        assert set(["OFFSET", "POSITION", "TYPE", 
+                    "DISH_DIAMETER", "FLAG_ROW", "MOUNT", "NAME", 
+                    "STATION"]) == set(_anttab.colnames()), "Measurement set conformance error"
+        self.antoffset = _anttab.getcol("OFFSET")
```

Comment:
May I suggest replacing these very verbose blocks by one-liners such as:
```
self._ant_columns = { name: _anttab.getcol(name) for name in _anttab.colnames() }
```
It's more elegant and succinct, and pollutes the data_handler's namespace less. And when you need to use these values, rather than thinking, "now which variable did I assign this column to", you can just look up the column in the dict.

Reply:
Actually I don't even see where you use this information, did you forget to ``git add`` something? But if you're just copying columns across to caltables wholesale, then I **definitely** suggest the succint programmatic approach.

Also, does this need to be done in ``__init__()``, could it not be done when you start writing the tables? The I/O process keeps an open ``handler.ms`` object throughout (until ``handler.unlock()`` is called when the final tile has been saved), so you could do it in the I/O process, perhaps when the first tile is being read in? That way it's concurrent with the main processing, rather than a delay on startup.

Reply:
resolved - the handles for the subtables aren't kept open - and actually they are never released properly. Shouldn't these handles be in __enter__ and __exit__ blocks to ensure they are released properly if there is a crash?

Reply:
But if there's a crash, we exit anyway and everything gets released, no?